unbound.in

```diff
@@ -42,12 +43,21 @@ newconf="# Generated by resolvconf$NL"
 for d in $DOMAINS; do
 	dn="${d%%:*}"
 	ns="${d#*:}"
+	create_unbound_insecure=false
+	create_unbound_private=false
 	case "$unbound_insecure" in
 	[Yy][Ee][Ss]|[Tt][Rr][Uu][Ee]|[Oo][Nn]|1)
-		newconf="$newconf${NL}server:$NL"
-		newconf="$newconf	domain-insecure: \"$dn\"$NL"
-		;;
+		create_unbound_insecure=true ;;
+	esac
+	case "$unbound_private" in
+	[Yy][Ee][Ss]|[Tt][Rr][Uu][Ee]|[Oo][Nn]|1)
+		create_unbound_private=true ;;
 	esac
+	if [[ "$create_unbound_insecure" = true || "$create_unbound_private" = true ]]; then
+		newconf="$newconf${NL}server:$NL"
+		[ "$create_unbound_insecure" = true ] && newconf="$newconf	domain-insecure: \"$dn\"$NL"
+		[ "$create_unbound_private" = true ] && newconf="$newconf	private-domain: \"$dn\"$NL"
```

Comment:
It's better to write `if ...; then ... fi` than use `&&` because if we ever use `set -e`, this will trip up.